### ISSUE TITLE
scapy.AsyncSniffer loss UDP packets and can only get 1514 bytes somehow

### ISSUE THREAD

**Issue by timmy8526** [secdev/scapy#2882]
#### Environment

-   Scapy version: scapy-2.4.4
-   Python version: 3.6.5
-   Operating System: Windows 10

#### How to reproduce

```
    def sniff_async(self, filter_str, stop_filter=None, timeout=10):
        sniffer = scapy.AsyncSniffer(
            iface=self.scapy_ifname,
            filter=filter_str,
            stop_filter=stop_filter,
            timeout=timeout)

        return sniffer

    def _filter_response(self, packet):
        print(len(packet))
            return False

    def get_info:
        # Start async sniff
        sniffer = self._local_host.sniff_async(
            filter_str='dst port 8097',
            stop_filter=self._filter_response,
            timeout=10)
        sniffer.start()

        # Wait sniffer started
        while not sniffer.running:
            time.sleep(1)

        # Send search packet (Broadcast)
        scapy.sendp( ... )

        # Wait sniff finished
        while sniffer.running:
            LOGGER.info('Qfinder: Waiting for respnse')
            time.sleep(1)

        if not sniffer.results:
            LOGGER.info('Qfinder: Timeout for sniffing target packets')
            raise TimeoutError

        return True
```

#### Actual result

The _filter_response first line will print out the length of packets.
And it always got 1514 bytes only ..

#### Expected result

Use wireshark to sniff the same packets and the length of packets were all greater than 1514 bytes.
(mostly 3000+ packets)

Is this a limitation of scapy? Or is there any config I didn't set up correctly to get the full length of packets?
Please help me to get the full packets ..

**Comment by gpotter2** [comment]
Wireshark is likely reassembling the packets.
Have a closer look at the packets you've dissected

**Comment by gpotter2** [comment]
See also https://scapy.readthedocs.io/en/latest/usage.html?highlight=performance#performance-of-scapy

**Comment by timmy8526** [comment]
Thanks @gpotter2 !

I did check the dissected packets, and found out that the problem is the filter_str (BPF) will skip the UDP packets I need.
I have revised the filter from "dst port 8097" to "udp", and now I could get all the rest of UDP packets!

However, to reassemble the packets, I use 'session=scapy.IPSession' which did help me to get the whole packets I want, but there's a problem .. the sniff results will return both [reassembled packet] and [non-reassembled packets].
Is there's a way I could only get the reassembled packet but ignore those non-reassembled packets?
